src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -598,8 +626,55 @@ void ITranslatable.Translate(ITranslator translator)
             translator.Translate(ref _projectStateAfterBuild, ProjectInstance.FactoryForDeserialization);
             translator.Translate(ref _savedCurrentDirectory);
             translator.Translate(ref _schedulerInducedError);
-            translator.TranslateDictionary(ref _savedEnvironmentVariables, StringComparer.OrdinalIgnoreCase);
-            translator.TranslateEnum(ref _buildRequestDataFlags, (int)_buildRequestDataFlags);
+
+            // This is a work-around for the bug https://github.com/dotnet/msbuild/issues/10208
+            // We are adding a version field to this class to make the ResultsCache backwards compatible with at least 2 previous releases.
```

Comment:
could you please let me know, what is going to happen if the newer version of the msbuild tries to deserialize the cache from the release older that 2 previous?

Reply:
We were told that support of 2 previous versions is good enough. So, we are allowed to crash/disregard the entries. We will need to compare versions to show a meaningful warning/error, but that's with the next PRs. With the cache that has no version, our only option will be most probably a crash.

Reply:
We will need to document which versions of caches we support.